python/src/BernsteinCopulaFactory_doc.i.in

```diff
@@ -12,66 +26,91 @@ DistributionFactory, EmpiricalBernsteinCopula"
-
     build(*sample, m*)
 
+    build(*sample, method, f*)
+
 Parameters
 ----------
-sample : 2-d sequence of float, of dimension *d*
-    The sample of size :math:`n>0` from which the copula is estimated.
+sample : 2-d sequence of float, of dimension :math:`d`
+    The sample of size :math:`\sampleSize>0` from which the copula is estimated.
 method : str
-    The name of the bin number selection method. Possible choices are *AMISE*, *LogLikelihood* and *PenalizedCsiszarDivergence*. Default is *LogLikelihood*, given by the *'BernsteinCopulaFactory-BinNumberSelection'* entry of :class:`~openturns.ResourceMap`.
-m : int
-    The number of sub-intervals in which all the edges of the unit cube
+    The name of the bin number selection method. Possible choices are *AMISE*, *LogLikelihood* and *PenalizedCsiszarDivergence*.
+    
+    Default is *LogLikelihood*.
+f : :class:`~openturns.Function`
+    The function defining the Csiszar divergence of interest used by
+    the  *PenalizedCsiszarDivergence* method.
+    
+    Default is Function().
+m : int,:math:`1 \leq m \leq \sampleSize`,
+    The bin number, i.e. the number of sub-intervals in which all the edges of the unit cube
     :math:`[0, 1]^d` are regularly partitioned.
+    
+    Default value is the value computed from the default bin number selection method.
 
 Returns
 -------
 copula : :class:`~openturns.Distribution`
-    The estimated copula as a generic distribution.
+    The empirical Bernstein copula as a generic distribution.
 
+Notes
+-----
+If the bin number :math:`m` is specified and does not divide the sample size :math:`\sampleSize`, then a part of the sample is
+removed for the result to be a copula. See :class:`~openturns.EmpiricalBernsteinCopula`.
 "
 // ---------------------------------------------------------------------
 
 %feature("docstring") OT::BernsteinCopulaFactory::buildAsEmpiricalBernsteinCopula
-"Build the nonparametric Bernstein copula estimator based on the empirical copula.
+"Build the empirical Bernstein copula as a native distribution.
 
 **Available usages**:
 
     buildAsEmpiricalBernsteinCopula()
 
     buildAsEmpiricalBernsteinCopula(*sample*)
 
-    buildAsEmpiricalBernsteinCopula(*sample, method, objective*)
-
     buildAsEmpiricalBernsteinCopula(*sample, m*)
 
+    buildAsEmpiricalBernsteinCopula(*sample, method, f*)
+
 Parameters
 ----------
 sample : 2-d sequence of float, of dimension *d*
-    The sample of size :math:`n>0` from which the copula is estimated.
+    The sample of size :math:`\sampleSize>0` from which the copula is estimated.
 method : str
     The name of the bin number selection method. Possible choices are *AMISE*, *LogLikelihood* and
-    *PenalizedCsiszarDivergence*. Default is *LogLikelihood*, given by the *'BernsteinCopulaFactory-BinNumberSelection'*
-    entry of :class:`~openturns.ResourceMap`.
-m : int
-    The number of sub-intervals in which all the edges of the unit cube
+    *PenalizedCsiszarDivergence*.
```

Comment:
so the list of possible values is given here. It is up to you to recall it in the description of the ResourceMap keys.